tool/tctl/common/autoupdate_command.go

```diff
@@ -0,0 +1,226 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/coreos/go-semver/semver"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/client/webclient"
+	autoupdatev1pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/autoupdate/v1"
+	"github.com/gravitational/teleport/api/types/autoupdate"
+	"github.com/gravitational/teleport/lib/auth/authclient"
+	"github.com/gravitational/teleport/lib/service/servicecfg"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// versionResponse is structure for formatting the autoupdate version response.
+type versionResponse struct {
+	TargetVersion string `json:"target_version"`
+}
+
+// AutoUpdateCommand implements the `tctl autoupdate` command for managing
+// autoupdate process for tools and agents.
+type AutoUpdateCommand struct {
+	app    *kingpin.Application
+	config *servicecfg.Config
+
+	configureCmd *kingpin.CmdClause
+	getCmd       *kingpin.CmdClause
+
+	mode               string
+	toolsTargetVersion string
+	proxy              string
+	format             string
+
+	// stdout allows to switch standard output source for resource command. Used in tests.
+	stdout io.Writer
+}
+
+// Initialize allows AutoUpdateCommand to plug itself into the CLI parser.
+func (c *AutoUpdateCommand) Initialize(app *kingpin.Application, config *servicecfg.Config) {
+	c.app = app
+	c.config = config
+	autoUpdateCmd := app.Command("autoupdate", "Teleport auto update commands.")
+
+	clientToolsCmd := autoUpdateCmd.Command("client-tools", "Client tools auto update commands.")
+
+	c.configureCmd = clientToolsCmd.Command("configure", "Edit client tools auto update configuration.")
```

Comment:
Do you have any suggestions for this one? Should I rename the command or update the help message?
I think `client-tools add` `client-tools update` might be confusing for user since this is singleton resource for configuration auto-updates, we also have `sso configure` with similar naming.
This command is added as a helper to manage resources related to client tool auto-updates. It will create the resource if it doesn't exist, or update it if it does https://github.com/gravitational/teleport/pull/39805/files#diff-8d3c0408dd826f27e9f13cd15f09f4d7e89202862f0b13ce131a2e3072c2a40dR257
I prefer previous naming `update` to be honest, but underlying behavior with resources must be `upsert`

Reply:
I've rename it to `get/set`

```
tctl autoupdate client-tools get
tctl autoupdate client-tools set --mode=enabled --target-version=17.0.0-dev
```
another option might be
```
tctl autoupdate client-tools enabled
tctl autoupdate client-tools disabled
tctl autoupdate client-tools set --target-version=17.0.0-dev
```
or
```
tctl autoupdate client-tools get
tctl autoupdate client-tools --set-mode=enabled --set-target-version=17.0.0-dev
```

Reply:
For the user might be confusing, since we update 2 resources at once if we define both flags for `--set-mode` AutoUpdateConfig, and `--set-target-version` AutoUpdateVersion, lets say I previously created AutoUpdateConfig with mode `disabled`, but in next command I type something like 
```
tctl autoupdate create --set-mode=enabled --set-target-version=17.0.0
```
it must fail because AutoUpdateConfig is already created

Reply:
CC: @hugoShaka